### ISSUE TITLE
Put switch category list into documentation

### ISSUE THREAD

**Issue by reznikmm** [alire-project/alire#1767]
Current Alire documentation says:
> It is also possible to specify a custom list of compiler switches for a category:

But it doesn't provide a list of such categories.

I consider that it worth to have such a list in the documentation. (Perhaps with default switches in each category per profile.)

**Comment by Fabien-Chouteau** [comment]
The list is available in the "Catalog format specification" under "build-switches: optional table of build profile switches definitions."

**Comment by kanigsson** [comment]
I agree that a documentation for the compilation switches for each profile is missing.
